archunit/src/test/java/com/tngtech/archunit/lang/ArchConditionTest.java

```diff
@@ -226,6 +234,41 @@ public void double_never_or() {
         assertThat(events.containViolation()).as("Events contain violation").isFalse();
     }
 
+    @Test
+    public void from_predicate() {
+        JavaClass object = new ClassFileImporter().importClass(Object.class);
+
+        assertThat(ArchCondition.from(alwaysFalse().as("some description")))
+                .hasDescription("some description")
+                .checking(object)
+                .containViolations(String.format("Class <%s> does not satisfy some description in (%s.java:0)", Object.class.getName(), Object.class.getSimpleName()));
+
+        assertThat(ArchCondition.from(alwaysTrue().as("some description")))
+                .checking(object).containNoViolation();
+
+        assertThat(not(ArchCondition.from(alwaysTrue().as("some description")).as("overwritten")))
+                .hasDescription("not overwritten")
+                .checking(object)
+                .containViolations(String.format("Class <%s> satisfies some description in (%s.java:0)", Object.class.getName(), Object.class.getSimpleName()));
```

Comment:
I have not seen many uses of overriding descriptions and not used that feature myself so I cannot judge on the expectation but I guess this is nothing unique to the Archrule-from-predicate but rather exists for all Archrules anyway.

Reply:
Yes, I just added that test because we have a custom `ArchCondition` where `as` is overridden. The test just checks that it behaves like all the other places. I used `as` myself in some cases to a) use a different language or b) make the intention clearer if the rule is very complex...